flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2326,6 +2340,52 @@ mlir::Value IntrinsicLibrary::genModulo(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, mustAddP, remPlusP, remainder);
 }
 
+// MVBITS
+void IntrinsicLibrary::genMvbits(llvm::ArrayRef<fir::ExtendedValue> args) {
+  // A conformant MVBITS(FROM,FROMPOS,LEN,TO,TOPOS) call satisfies:
+  //     FROMPOS >= 0
+  //     LEN >= 0
+  //     TOPOS >= 0
+  //     FROMPOS + LEN <= BIT_SIZE(FROM)
+  //     TOPOS + LEN <= BIT_SIZE(TO)
+  // MASK = -1 >> (BIT_SIZE(I) - LEN)
+  // TO = LEN == 0 ? TO : ((!(MASK << TOPOS)) & TO) |
+  //                      (((FROM >> FROMPOS) & MASK) << TOPOS)
+  assert(args.size() == 5);
+  auto unbox = [&](fir::ExtendedValue exv) {
+    auto arg = exv.getUnboxed();
+    assert(arg && "nonscalar mvbits argument");
+    return *arg;
+  };
```

Comment:
I feel like we had a tiny function that does this somewhere, but I can't find it. Looks like with use the `if (auto *...` idiom instead.